src/frontier_exploration/src/frontier_rrt_search.cpp

```diff
@@ -0,0 +1,316 @@
+#include <ros/ros.h>
+#include <tf/tf.h>
+#include <tf2_ros/transform_listener.h>
+#include <tf2_ros/transform_broadcaster.h>
+
+#include "frontier_rrt_search.h"
+
+FrontierRRTSearch::FrontierRRTSearch(ros::NodeHandle &nh) : nh_(nh)
+{
+
+    ns = ros::this_node::getName();
+    ros::param::param<float>(ns + "/eta", eta, 10);
+    ros::param::param<std::string>(ns + "/map_topic", map_topic, "/map");
+    ros::param::param<std::string>(ns + "/robot_leader", robot_leader, "agent1");
+
+    map_sub = nh_.subscribe(map_topic, 100, &FrontierRRTSearch::mapCallBack, this);
+    targets_pub = nh_.advertise<geometry_msgs::PointStamped>("/detected_points", 10);
```

Comment:
Always add new topics in your node namespace something like "/task_allocator/detected_points" so that when the full system is running we dont have to check whos publishing each topic